src/app/shared/components/add-station/add-station.component.ts

```diff
@@ -0,0 +1,119 @@
+  private configSettings: Config;
+  private modalElement: any;
+  public agencies: Array<Agency>;
+  public stationTypes: Array<StationType>;
+  public modalSubscription: any;
+  public modalRef;
+  public addStationForm: FormGroup;
+
+
+  constructor(private _nssService: NSSService, private _modalService: NgbModal, private _fb: FormBuilder,
+    private _settingsService: SettingsService, private _configService: ConfigService, private _toasterService: ToasterService,
+    private _authService: AuthService) {
+    this.addStationForm = _fb.group({
+      code : new FormControl(null, Validators.required),
+      agencyID : new FormControl(null, Validators.required),
+      name : new FormControl(null),
+      stationTypeID : new FormControl( null, Validators.required),
+      latitude : new FormControl(null, [Validators.min(-90), Validators.max(90), Validators.required] ),
+      longitude : new FormControl(null, [Validators.min(-180), Validators.max(180), Validators.required]),
+
+    });
+
+    this.configSettings = this._configService.getConfiguration();
+  }
+
+  ngOnInit() {
+    this.modalSubscription = this._nssService.showAddStationModal.subscribe((show: boolean) => {
+      if (show) { this.showModal(); }
+    });
+    this.modalElement = this.addStationModal;
+
+    this._nssService.agencies.subscribe((agencyList: Array<Agency>) => {
+      this.agencies = agencyList;
+    });
+    this._nssService.stationTypes.subscribe((stationTypeList: Array<StationType>) =>{
+      this.stationTypes = stationTypeList;
+    })
+  }
+
+  public showModal(): void {
+    
+    this.modalRef = this._modalService.open(this.modalElement, { backdrop: 'static', keyboard: false, size: 'lg' });
+    this.modalRef.result.then(
+        result => {
+            // this is the solution for the first modal losing scrollability
+            if (document.querySelector('body > .modal')) {
+                document.body.classList.add('modal-open');
+            }
+            
+        
+        }
+    );
+  }
+  
+  public clearStation() {
+    this.addStationForm.reset();
+
+  }
+
+  public submitStation() {
+    const location = {type: 'Point', coordinates: [ this.addStationForm.get('latitude').value, this.addStationForm.get('longitude').value]}
+    
+    let station = JSON.parse(JSON.stringify(this.addStationForm.value));
+    delete station.latitude;
+    delete station.longitude;
+    station = {...station, 'location': location}
+     /*
+      STATION OBJECT EXAMPLE Format
+      {
+        "code":"11111111",
+        "agencyID": 1,
+        "name":"example station 1 at example location, ex. river",
+        "stationTypeID":"1",
+        "location": {
+            "type": "Point",
+            "coordinates": [
+                "0": -99.999,
+                "1": 44.444
+            ]
+        }
+    }*/
+
+    //console.log('station: ', station);
+    this._settingsService.postEntityGageStats(station, 'stations')
```

Comment:
I think this is fine because it's a self-explanatory endpoint, but in most of the app we try to use the config to get these urls